### PR TITLE
Task/map/load restaurants locations

### PR DESCRIPTION
**Pull request by teamagam7**

### CODE REVIEW

File: HeverTasteApp/app/src/main/java/com/tm7/hevertaste/RestaurantMapActivity.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+                .findFragmentById(R.id.map);
+        mapFragment.getMapAsync(this);
+
+        DownloadKmlTask downloadKmlTask = new DownloadKmlTask(RestaurantMapActivity.this);
+        downloadKmlTask.execute();
+    }
+
+    @Override
+    public void onMapReady(GoogleMap googleMap) {
+
```
Comment (teamagam3) [inline HeverTasteApp/app/src/main/java/com/tm7/hevertaste/RestaurantMapActivity.java]: Remove redundant space

File: HeverTasteApp/app/src/main/java/com/tm7/hevertaste/RestaurantMapActivity.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        RestaurantsKmlImporter temp = new RestaurantsKmlImporter();
```
Comment (teamagam3) [inline HeverTasteApp/app/src/main/java/com/tm7/hevertaste/RestaurantMapActivity.java]: Change variable name

File: HeverTasteApp/app/src/main/java/com/tm7/hevertaste/RestaurantMapActivity.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        List<Restaurant> restaurants = temp.getRestaurants();
+        for (Restaurant restaurant : restaurants) {
```
Comment (teamagam3) [inline HeverTasteApp/app/src/main/java/com/tm7/hevertaste/RestaurantMapActivity.java]: Extract to `loadRestaurants()`

File: HeverTasteApp/app/src/main/java/com/tm7/hevertaste/RestaurantsKmlImporter.java [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+import java.util.List;
+
+import javax.inject.Inject;
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+
+public class RestaurantsKmlImporter implements RestaurantsProvider {
+
+    private final String KML_LOCATION = "storage/emulated/0/Android/data/com.tm7.hevertaste/files/restaurants.kml";
```
Comment (teamagam3) [inline HeverTasteApp/app/src/main/java/com/tm7/hevertaste/RestaurantsKmlImporter.java]: Extract to config

File: HeverTasteApp/app/src/main/java/com/tm7/hevertaste/RestaurantsKmlImporter.java [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+    private final String RESTAURANT_TAG_NAME = "Placemark";
```
Comment (teamagam3) [inline HeverTasteApp/app/src/main/java/com/tm7/hevertaste/RestaurantsKmlImporter.java]: Those aren't important for config.
The fields here are used for internal logic.

File: HeverTasteApp/app/src/main/java/com/tm7/hevertaste/RestaurantsKmlImporter.java [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+                    mRestaurants.add(elementToRestaurants(element));
+                }
+            }
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    private Restaurant elementToRestaurants (Element element) {
+        String name =
```
Comment (teamagam3) [inline HeverTasteApp/app/src/main/java/com/tm7/hevertaste/RestaurantsKmlImporter.java]: Extract to methods: `extractName()`, `extractDescription()` and `extractCoordinates()`
If you find better names , that's fine.

File: HeverTasteApp/app/src/main/java/com/tm7/hevertaste/RestaurantsKmlImporter.java [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+
+    private NodeList getPlacemarks()  {
+        Document document = null;
+        try {
+            File kmlFile = new File(KML_LOCATION);
+            DocumentBuilderFactory documentBuilderFactory = DocumentBuilderFactory.newInstance();
+            DocumentBuilder documentBuilder = documentBuilderFactory.newDocumentBuilder();
+            document = documentBuilder.parse(kmlFile);
+            document.getDocumentElement().normalize();
+        } catch (ParserConfigurationException e) {
```
Comment (teamagam3) [inline HeverTasteApp/app/src/main/java/com/tm7/hevertaste/RestaurantsKmlImporter.java]: Don't you need to do something with the exceptions?

File: HeverTasteApp/app/src/main/java/com/tm7/hevertaste/RestaurantMapActivity.java (hunk 1 above)
Comment (teamagam7) [inline HeverTasteApp/app/src/main/java/com/tm7/hevertaste/RestaurantMapActivity.java]: done

File: HeverTasteApp/app/src/main/java/com/tm7/hevertaste/RestaurantMapActivity.java (hunk 2 above)
Comment (teamagam7) [inline HeverTasteApp/app/src/main/java/com/tm7/hevertaste/RestaurantMapActivity.java]: done

File: HeverTasteApp/app/src/main/java/com/tm7/hevertaste/RestaurantMapActivity.java (hunk 3 above)
Comment (teamagam7) [inline HeverTasteApp/app/src/main/java/com/tm7/hevertaste/RestaurantMapActivity.java]: done

File: HeverTasteApp/app/src/main/java/com/tm7/hevertaste/RestaurantsKmlImporter.java (hunk 6 above)
Comment (teamagam7) [inline HeverTasteApp/app/src/main/java/com/tm7/hevertaste/RestaurantsKmlImporter.java]: done

**Pull request merged**